### PR TITLE
feat(remix-dev/cli/init): add `--no-delete` flag

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Follow-up of #3571

### CODE REVIEW

File: packages/remix-dev/cli/commands.ts [hunk 1]
```diff
@@ -56,7 +56,13 @@ export async function create({
   spinner.clear();
 }
 
-export async function init(projectDir: string) {
+type InitFlags = {
+  deleteScript?: boolean;
```
Comment (machour) [inline packages/remix-dev/cli/commands.ts:60]: just for clarity, should this be renamed to `deleteInitScript` ?

File: packages/remix-dev/cli/run.ts [hunk 2]
```diff
@@ -376,7 +379,7 @@ export async function run(argv: string[] = process.argv.slice(2)) {
       if (hasInitScript) {
         if (installDeps) {
           console.log("💿 Running remix.init script");
-          await commands.init(projectDir);
+          await commands.init(projectDir, { deleteScript: true });
```
Comment (machour) [inline packages/remix-dev/cli/run.ts:382]: This isn't really needed as this is the default

**Review by machour — APPROVED** [review]

File: packages/remix-dev/cli/commands.ts (hunk 1 above)
Comment (MichaelDeBoey) [reply packages/remix-dev/cli/commands.ts:60]: We're calling the `init` command, so it's obvious that we're talking about the init script I think.

File: packages/remix-dev/cli/run.ts (hunk 2 above)
Comment (MichaelDeBoey) [reply packages/remix-dev/cli/run.ts:382]: Indeed not necessary, but I like the explicitness.

**Review by mcansh — APPROVED** [review]

**Review by kentcdodds — APPROVED** [review]
Super! Thanks 👍

**Comment by kentcdodds** [comment]
Rats. I took too long reviewing. Could you handle the merge conflict?

**Review by kentcdodds — APPROVED** [review]
Thanks!

**Pull request merged**